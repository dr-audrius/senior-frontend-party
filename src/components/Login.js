import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import { PostData } from '../services/PostData';

class Login extends Component {

  constructor() {
    super();

    this.state = {
      username: '',
      password: '',
      redirectToReferrer: false,

    };

    this.login = this.login.bind(this);
    this.onChange = this.onChange.bind(this);
  }

  componentWillMount() {
    document.body.className = "body-component-login";
  }

  login() {

    let URL = 'http://playground.tesonet.lt/v1/tokens';

    if (this.state.username && this.state.password) {
      PostData(URL, this.state).then((result) => {
        let responseJson = result;

        if (responseJson.token) {

          localStorage.setItem('token', JSON.stringify(responseJson));
          sessionStorage.setItem('userData', JSON.stringify(responseJson));

          this.setState({ redirectToReferrer: true });
        }
      });
    }
  }

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  render() {

    if (this.state.redirectToReferrer) {
      return (<Redirect to={'/'} />)
    }

    if (localStorage.getItem('token')) {
      return (<Redirect to={'/'} />)
    }

    return (

      <div className="login">
        <h2></h2>

        <div className="input-group mb-3">
          <div className="input-group-prepend">
            <span className="input-group-text" id="basic-addon1"><i className="fa fa-user"></i></span>
          </div>
          <input type="text" name="username" placeholder="Username" className="form-control" onChange={this.onChange} />
        </div>

        <div className="input-group mb-3">
          <div className="input-group-prepend">
            <span className="input-group-text" id="basic-addon1"><i className="fa fa-lock"></i></span>
          </div>
          <input type="password" name="password"  placeholder="Password" className="form-control" onChange={this.onChange}/>
        </div>

        <input type="submit" className="btn btn-primary" value="Login" onClick={this.login} />

      </div>

    );
  }
}
export default Login;   
