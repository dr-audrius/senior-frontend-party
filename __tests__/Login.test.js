import React from 'react';
//import ReactDOM from 'react-dom';
import Login from '../src/components/Login';
import renderer from 'react-test-renderer';

test(" snapshot test", () => {
    const tree =renderer.create(<Login />).toJSON();
    console.log(tree);
    expect(tree).toMatchSnapshot();
})