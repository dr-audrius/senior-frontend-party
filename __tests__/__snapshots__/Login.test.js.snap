// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` snapshot test 1`] = `
<div
  className="login"
>
  <h2 />
  <input
    className="form-control"
    name="username"
    onChange={[Function]}
    placeholder="Username"
    type="text"
  />
  <input
    className="form-control"
    name="password"
    onChange={[Function]}
    placeholder="Password"
    type="password"
  />
  <input
    className="btn btn-primary"
    onClick={[Function]}
    type="submit"
    value="Login"
  />
</div>
`;
